app/javascript/packages/document-capture/hooks/use-step-logger.ts

```diff
@@ -0,0 +1,34 @@
+import { useCallback, useContext, useEffect } from 'react';
+import AnalyticsContext from '../context/analytics';
+import InPersonLocationStep from '../components/in-person-location-step';
+import InPersonPrepareStep from '../components/in-person-prepare-step';
+import InPersonSwitchBackStep from '../components/in-person-switch-back-step';
+
+export const LOGGED_STEPS: string[] = [
+  InPersonLocationStep.stepName,
+  InPersonPrepareStep.stepName,
+  InPersonSwitchBackStep.stepName,
+];
+
+const isLoggedStep = (stepName?: string): boolean => !!stepName && LOGGED_STEPS.includes(stepName);
+
+function useStepLogger(currentStep?: string) {
+  const { trackEvent } = useContext(AnalyticsContext);
+  const onStepSubmit = useCallback(
+    (stepName?: string) => {
+      if (isLoggedStep(stepName)) {
+        trackEvent(`IdV: ${stepName} submitted`);
```

Comment:
Re-reading the ticket, we want to log the selected location for a post office on the locations screen. It would seem like something we could include in this `submitted` event, though with how this is currently implemented, making that metadata available here could involve some extra work.

I initially planned to address this separately in a follow-on pull request, but I'd wonder if it might have some fundamental impact on how we approach this implementation.

Reply:
This was revised in d46eda7be.